src/latency.c

```diff
@@ -490,6 +490,147 @@ sds createLatencyReport(void) {
 
 /* ---------------------- Latency command implementation -------------------- */
 
+/* An array of time buckets, each representing a latency range,
+ * between 1 nanosecond and roughly 1 second.
+ * Each bucket covers twice the previous bucket s range.
+ * Empty buckets are not printed.
+ * Everything above 1sec is considered +Inf. */
+sds fillCommandCDF(sds info, const char* histogram_name, struct hdr_histogram* histogram){
+    info = sdscatprintf(info, "latency_hist_usec_%s:calls=%lld,histogram=[",
+        histogram_name, (long long) histogram->total_count);
+    struct hdr_iter iter;
+    hdr_iter_log_init(&iter, histogram, 1024,2);
+    size_t bucket_pos = 0;
+    int64_t previous_count = 0;
+    while (hdr_iter_next(&iter)) {
+        const int64_t micros = iter.highest_equivalent_value / 1000;
+        const int64_t cumulative_count = iter.cumulative_count;
+        if(cumulative_count > previous_count){
+            if (bucket_pos>0)
+                info = sdscatprintf(info,";");
+            info = sdscatprintf(info,"(%lld:%lld)", (long long) micros, (long long) cumulative_count);
+            bucket_pos++;
+        }
+        previous_count = cumulative_count;
+
+    }
+    info = sdscatprintf(info,"]\r\n");
+    return info;
+}
+
+/* An array of time buckets, each representing a latency range,
+ * between 1 nanosecond and roughly 1 second.
+ * Each bucket covers twice the previous bucket s range.
+ * Empty buckets are not printed.
+ * Everything above 1sec is considered +Inf. */
+void fillCommandCDFResp(client *c, const char* histogram_name, struct hdr_histogram* histogram){
+    addReplyArrayLen(c,5);
+    addReplyBulkCString(c,histogram_name);
+    addReplyBulkCString(c,"calls");
+    addReplyLongLong(c,(long long) histogram->total_count);
+    addReplyBulkCString(c,"histogram");
+    void *replylen = addReplyDeferredLen(c);
```

Comment:
Deferred reply has overheads (small packets, and lots of write syscalls).
is it possible to predict the size of the array easily? or even make a quick counting?

Reply:
@oranagra to predict the array/map len we would need to go through the histogram iterator twice given we don't output the empty buckets. at max we will have 30 buckets. do you think we should do it?

Reply:
He alternative is that each command has its own write system call and it's own packet.. 
Unless #9934 will help. 
We can keep the simple form, and maybe optimize in the future.. 
Is suppose that this command doesn't require high efficient anyway, so shorter code is preferred.

Reply:
Based on: 

>  Is suppose that this command doesn't require high efficient anyway, so shorter code is preferred.

and 

>  We can keep the simple form, and maybe optimize in the future..

I'll add a note on the PR description. Agree that for now the simpler the better